File: verify_comments.py
import os, re, sys
sys.stdout.reconfigure(encoding='utf-8', errors='replace')

COMMENT_LINE = re.compile(r'^\s*//')
COMMENT_INLINE = re.compile(r'\)\s*//|\]\s*//|\}\s*//|,\s*//')

dart_dir = 'c:/laragon/www/LUNGO/frontend/app/lib'
total_comments = 0
files_with_comments = []

for root, dirs, files in os.walk(dart_dir):
    for fname in files:
        if not fname.endswith('.dart'):
            continue
        fpath = os.path.join(root, fname)
        file_count = 0
        with open(fpath, 'r', encoding='utf-8', errors='ignore') as f:
            for i, line in enumerate(f, 1):
                if COMMENT_LINE.search(line) or COMMENT_INLINE.search(line):
                    total_comments += 1
                    file_count += 1
        if file_count > 0:
            rel = os.path.relpath(fpath, dart_dir)
            files_with_comments.append(f'{rel}: {file_count} komentar')

print(f'Total komentar tersisa: {total_comments}')
if files_with_comments:
    print('File dengan komentar:')
    for f in files_with_comments:
        print(f'  {f}')
else:
    print('BERSIH SEMPURNA - 0 komentar!')

File: verify_comments.js
const fs = require('fs');
const path = require('path');

function getAllDartFiles(dir) {
  let files = [];
  const items = fs.readdirSync(dir);
  for (const item of items) {
    const fullPath = path.join(dir, item);
    if (fs.statSync(fullPath).isDirectory()) {
      files = files.concat(getAllDartFiles(fullPath));
    } else if (item.endsWith('.dart')) {
      files.push(fullPath);
    }
  }
  return files;
}

const dartDir = 'c:/laragon/www/LUNGO/frontend/app/lib';
const files = getAllDartFiles(dartDir);
let total = 0;
const found = [];

for (const file of files) {
  const lines = fs.readFileSync(file, 'utf8').split('\n');
  lines.forEach((line, idx) => {
    if (line.includes('//') &&
        !line.includes('http://') &&
        !line.includes('https://') &&
        !line.includes('dana://')) {
      total++;
      found.push(path.relative(dartDir, file) + ':' + (idx+1) + ' -> ' + line.trim());
    }
  });
}

console.log('Komentar non-URL tersisa: ' + total);
found.slice(0, 30).forEach(f => console.log(f));
if (total === 0) console.log('BERSIH SEMPURNA!');
